specs/eip4844/p2p-interface.md

```diff
@@ -65,34 +78,35 @@ The new topics along with the type of the `data` field of a gossipsub message ar
 
-EIP-4844 introduces a new global topic for beacon block and blobs-sidecars.
+EIP-4844 introduces new global topics for blob sidecars.
 
 ##### `beacon_block`
 
-This topic is deprecated and clients **MUST NOT** expose in their topic set to any peer. Implementers do not need to do
-anything beyond simply skip implementation, and it is explicitly called out as it is a departure from previous versioning
-of this topic.
+The *type* of the payload of this topic changes to the (modified) `SignedBeaconBlock` found in EIP4844.
 
-Refer to [the section below](#transitioning-the-gossip) for details on how to transition the gossip.
+##### `blob_sidecar_{index}`
 
-##### `beacon_block_and_blobs_sidecar`
+This topic is used to propagate signed blob sidecars, one for each sidecar index.
 
-This topic is used to propagate new signed and coupled beacon blocks and blobs sidecars to all nodes on the networks.
+The following validations MUST pass before forwarding the `sidecar` on the network, assuming the alias `sidecar = signed_blob_sidecar.message`:
 
-In addition to the gossip validations for the `beacon_block` topic from prior specifications, the following validations MUST pass before forwarding the `signed_beacon_block_and_blobs_sidecar` on the network.
-Alias `signed_beacon_block = signed_beacon_block_and_blobs_sidecar.beacon_block`, `block = signed_beacon_block.message`, `execution_payload = block.body.execution_payload`.
-- _[REJECT]_ The KZG commitments correspond to the versioned hashes in the transactions list
-  -- i.e. `verify_kzg_commitments_against_transactions(block.body.execution_payload.transactions, block.body.blob_kzg_commitments)`
-
-Alias `sidecar = signed_beacon_block_and_blobs_sidecar.blobs_sidecar`.
-- _[IGNORE]_ the `sidecar.beacon_block_slot` is for the current slot (with a `MAXIMUM_GOSSIP_CLOCK_DISPARITY` allowance)
-  -- i.e. `sidecar.beacon_block_slot == block.slot`.
-- _[REJECT]_ The KZG commitments in the block are valid against the provided blobs sidecar
-  -- i.e. `validate_blobs_sidecar(block.slot, hash_tree_root(block), block.body.blob_kzg_commitments, sidecar)`
+- _[REJECT]_ The sidecar is for the correct topic --
+  i.e. `sidecar.index` matches the topic `{index}`.
+- _[IGNORE]_ The sidecar is not from a future slot (with a `MAXIMUM_GOSSIP_CLOCK_DISPARITY` allowance)
+- _[IGNORE]_ The sidecar is from a slot greater than the latest finalized slot --
+  i.e. validate that `sidecar.slot > compute_start_slot_at_epoch(state.finalized_checkpoint.epoch)`
+- _[REJECT]_ The proposer signature, `signed_blob_sidecar.signature`, is valid with respect to the `sidecar.proposer_index` pubkey.
+- _[IGNORE]_ The sidecar is the only sidecar with valid signature received for the tuple `(sidecar.slot, sidecar.proposer_index, sidecar.index)`.
+  -- Clients MUST discard blocks where multiple sidecars for the same proposer and index have been observed.
+- _[REJECT]_ The sidecar is proposed by the expected `proposer_index` for the block's slot
```

Comment:
I wonder if this should come before the proposer signature check. That is, abort if it's the wrong proposer before wasting the time on the signature check. This comes at the cost of going and looking for the block_parent_root, but that condition is already an MAY queue when you go look for it.

this is slightly different than the beacon_block conditions which have the proposer signature check before the shuffling dependency check. Beacon blocks are slashable so at most one can be made per validator per slot before being slashed. Whereas validators have an ~infinite allowance here with impunity.

That said, I question the ordering of the conditions beacon_block gossip spec... in the normal case you know the shuffling and should reject before doing the proposer signature check

Reply:
> I wonder if this should come before the proposer signature check.

we've never established a strict spec requirement to do the checks in the given order (spec grey zone), ergo clients today do checks in whatever order they like, depending on the state and implementation of internal caches / timings / etc - this has the downside of leading to mismatches in REJECT / IGNORE between clients for messages with multiple errors.

If we want to start mandating an order here, we need to explicitly state this (which would be outside the scope of this PR) - this would have an associated performance cost as well.

Reply:
Yeah, I suppose the order isn't mandated but we have tried to always put them in a natural/anti-dos order

Reply:
copy-pasting is what I do best: https://github.com/ethereum/consensus-specs/blob/dev/specs/phase0/p2p-interface.md#beacon_block